### PR TITLE
#25 Setup shared_pref and store authToken data

### LINKED ISSUE #25: [Backend] Setup local storage - shared preference
## Why
The application will need to store data to key/value shared pref.

## Acceptance Criteria
- Initialize shared_pref and equivalent in iOS.
- If possible, prepare encryption storage ready.

### PR DESCRIPTION
**Pull request by sleepylee**
Closes #25

## What happened 👀

- As described per title.
 
## Insight 📝

- Add shared_pref
- Store token data once login is successfully, later we will handle the behavior when login successfully.
 
## Proof Of Work 📹
image

### CODE REVIEW

File: lib/use_cases/login_use_case.dart [hunk 1]
```diff
@@ -15,17 +16,25 @@ class LoginCredential {
...
-        .then<Result<AuthToken>>((value) => Success(value))
+        .then((value) => _persistTokenData(value))
         .onError((err, stackTrace) => Failed(
             UseCaseException(NetworkExceptions.fromDioException(err), err)));
   }
+
+  Result<dynamic> _persistTokenData(AuthToken data) {
+    _sharedPreferencesStorage.saveTokenType(data.tokenType);
+    _sharedPreferencesStorage.saveAccessToken(data.accessToken);
+    _sharedPreferencesStorage.saveAccessToken(data.refreshToken);
```
Comment (markgravity) [inline lib/use_cases/login_use_case.dart:37]: It should be `saveRefreshToken`, right? 🤔

**Review by markgravity — CHANGES_REQUESTED** [review]

File: lib/use_cases/login_use_case.dart (hunk 1 above)
Comment (sleepylee) [reply lib/use_cases/login_use_case.dart:37]: ouch copy-paste issue 😭

**Review by markgravity — APPROVED** [review]
LGTM 🤖

File: lib/use_cases/login_use_case.dart [hunk 2]
```diff
@@ -15,17 +16,25 @@ class LoginCredential {
...
-        .then<Result<AuthToken>>((value) => Success(value))
+        .then((value) => _persistTokenData(value))
         .onError((err, stackTrace) => Failed(
             UseCaseException(NetworkExceptions.fromDioException(err), err)));
   }
+
+  Result<dynamic> _persistTokenData(AuthToken data) {
+    _sharedPreferencesStorage.saveTokenType(data.tokenType);
+    _sharedPreferencesStorage.saveAccessToken(data.accessToken);
+    _sharedPreferencesStorage.saveRefreshToken(data.refreshToken);
```
Comment (manh-t) [inline lib/use_cases/login_use_case.dart:37]: should we `await` here before return value? 🙏

**Review by manh-t — APPROVED** [review]
One minor concern, rest LGTM 🙏

File: lib/use_cases/login_use_case.dart (hunk 2 above)
Comment (sleepylee) [reply lib/use_cases/login_use_case.dart:37]: it's not an async function, but inside of each call, when calling to shared_pref `set` method it returns a Future. In this case, I think I should return await inside then? 🤔

**Pull request merged**